pkg/loki/modules.go

```diff
@@ -1626,10 +1625,11 @@ func (t *Loki) initMemberlistKV() (services.Service, error) {
 	dnsProvider := dns.NewProvider(util_log.Logger, dnsProviderReg, dns.GolangResolverType)
 
 	var err error
-	t.Cfg.MemberlistKV.AdvertiseAddr, err = GetInstanceAddr(
+	t.Cfg.MemberlistKV.AdvertiseAddr, err = ring.GetInstanceAddr(
 		t.Cfg.MemberlistKV.AdvertiseAddr,
-		t.Cfg.Common.InstanceInterfaceNames,
+		t.Cfg.Common.Ring.InstanceInterfaceNames,
```

Comment:
this line of change is a no-op as `Common.Ring.InstanceInterfaceNames` is set from `Common.InstanceInterfaceNames` unless the user has explicitly configured different values for them - not sure if doing that makes sense.

I only updated this as memberlist is the underlying transport for the ring, it makes sense for it to inherit ring settings rather than loki level settings.